### PR TITLE
Thermostat: check if we are attached before getting clientWidth

### PR DESCRIPTION
**Pull request by bramkragten**

### CODE REVIEW

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 1]
```diff
@@ -79,6 +79,9 @@ export class HuiThermostatCard extends hassLocalizeLitMixin(LitElement)
   private _roundSliderStyle?: TemplateResult;
   private _jQuery?: any;
   private _broadCard?: boolean;
+  private _loaded?: boolean;
+  private _updated?: boolean;
+  private _attached?: boolean;
```
Comment (thomasloven) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]: I think `this.isConnected` can be used instead.

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 2]
```diff
@@ -101,6 +104,14 @@ export class HuiThermostatCard extends hassLocalizeLitMixin(LitElement)
     this._config = { theme: "default", ...config };
   }
 
+  public connectedCallback(): void {
+    super.connectedCallback();
+    this._attached = true;
+    if (this._updated && !this._loaded) {
```
Comment (thomasloven) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]: This check and the one in `firstUpdated` can be combined and placed in `_initialLoad`. Not sure which is clearer.

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: I prefer to only call `firstUpdated` once

**Review by thomasloven — APPROVED** [review]

File: src/panels/lovelace/cards/hui-thermostat-card.ts (hunk 2 above)
Comment (thomasloven) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: Fair enough.

**Comment by balloob** [comment]
In the updated Lit Elemenet, all updates are deferred until we are connected: https://github.com/Polymer/lit-element/blob/master/CHANGELOG.md

**Pull request merged**